### PR TITLE
[Storage] `az storage copy` add new option `--cap-mbps`

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

Fix issue #17544
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py [hunk 1]
```diff
@@ -33,6 +34,8 @@ def storage_copy(source, destination, put_md5=None, recursive=None, blob_type=No
         flags.append('--content-type=' + content_type)
     if follow_symlinks is not None:
         flags.append('--follow-symlinks=true')
+    if cap_mbps is not None:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py:37]: could you  add test for the new flag?

Comment (PARADISSEEKR) [reply src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py:37]: Test has been added

File: src/azure-cli/azure/cli/command_modules/storage/_params_azure_stack.py [hunk 2]
```diff
@@ -634,6 +634,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('content_type', arg_group='Additional Flags', help="Specify content type of the file. ")
         c.argument('follow_symlinks', arg_group='Additional Flags', action='store_true',
                    help='Follow symbolic links when uploading from local file system.')
+        c.argument('cap_mbps', arg_group='Additional Flags', help="Caps the transfer rate, in megabits per second. "
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params_azure_stack.py:637]:
```suggestion
        c.argument('cap_mbps', arg_group='Additional Flags', help="Cap the transfer rate, in megabits per second. "
```

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 3]
```diff
@@ -964,6 +964,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('content_type', arg_group='Additional Flags', help="Specify content type of the file. ")
         c.argument('follow_symlinks', arg_group='Additional Flags', action='store_true',
                    help='Follow symbolic links when uploading from local file system.')
+        c.argument('cap_mbps', arg_group='Additional Flags', help="Caps the transfer rate, in megabits per second. "
+                   "Moment-by-moment throughput might vary slightly from the cap. "
+                   "If this option is set to zero, or it is omitted, the throughput isn't capped. ")
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:969]: Do we need a validator to check if it's valid number? At least it needs to be a number.

Comment (PARADISSEEKR) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py:969]: `azcopy` already has a validator to check, do we need to check it again?

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**